analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd

```diff
@@ -99,97 +107,126 @@ ctnnb1_e3_tsb <-
+  mutate(CTNNB1.Exon3 = Kids_First_Biospecimen_ID %in% ctnnb1_e3_tsb) %>%
+  mutate(BRAF.V600E = Kids_First_Biospecimen_ID %in% braf_v600e_tsb) %>%
   dplyr::arrange(Kids_First_Participant_ID, sample_id)
+
+# print this lesion table
+write_tsv(snv_lesion_df, output_lesions)
 ```
 
 ## Defining Molecular Subtype Table
 
 Append molecular subtypes defined as follows:
 1. `CRANIO, PAP`: patient has BRAF.V600E mutation and is older than the age_of_adulthood (40 years in this case)
 1. `CRANIO, ADAM`:patient has CTNNB1.Exon3 mutation and is younger than the age_of_adulthood (40 years in this case)
-1. `CRANIO, ADAM;CRANIO, PAP` Both of the above (this is actually impossible given the restriction on age)
 1. `CRANIO, To be classified` None of the above or DNA sample does not exist
 Select the columns from the table that the PIs have requested
 
 ```{r meta-subtype-table}
 # Dispute this all you like
 age_of_adulthood <- 40 * 365.25
 
-snv_subtype <-
-  snv_lesion_df %>%
+# SNV-related bs id subtypes
+snv_subtype <- snv_lesion_df %>%
   mutate(fourty_or_older = as.numeric(age_at_diagnosis_days) >= age_of_adulthood) %>%
   mutate(molecular_subtype =
            case_when(
              (BRAF.V600E & fourty_or_older) & !(CTNNB1.Exon3 & !fourty_or_older) ~ "CRANIO, PAP",
-             !(BRAF.V600E & fourty_or_older) & (CTNNB1.Exon3 & !fourty_or_older) ~ "CRANIO, ADAM",
-              !(BRAF.V600E & fourty_or_older) & (CTNNB1.Exon3) ~ "CRANIO, ADAM",
-             (BRAF.V600E & fourty_or_older) & (CTNNB1.Exon3 & !fourty_or_older) ~ "CRANIO, ADAM;CRANIO, PAP",
-             !(BRAF.V600E & fourty_or_older) & !(CTNNB1.Exon3 & !fourty_or_older) ~ "CRANIO, To be classified",
-            is.na(Kids_First_Biospecimen_ID_DNA) ~ "CRANIO, To be classified"
-            )) %>%
-  select(Kids_First_Participant_ID,sample_id,molecular_subtype)
-
-meta_subtype <-
-  full_join(allmeta,
-            snv_subtype,
-            by=c("Kids_First_Participant_ID","sample_id")) %>%
-  group_by(Kids_First_Participant_ID, sample_id, age_at_diagnosis_days) %>%
-     summarise(Kids_First_Biospecimen_ID_DNA = str_c(unique(Kids_First_Biospecimen_ID_DNA), collapse=","),
-               Kids_First_Biospecimen_ID_RNA = str_c(unique(Kids_First_Biospecimen_ID_RNA), collapse=","),
-               # Add this to combine and fix subtypes if subtyped as ADAM + To be classified
-               molecular_subtype = str_c(unique(molecular_subtype), collapse=";")) %>%
-  dplyr::mutate(molecular_subtype = case_when(molecular_subtype == "CRANIO, To be classified;CRANIO, ADAM" ~ "CRANIO, ADAM",
-                                              molecular_subtype == "CRANIO, ADAM;CRANIO, To be classified" ~ "CRANIO, ADAM",
-                                              TRUE ~ as.character(molecular_subtype))) %>%
-  dplyr::arrange(Kids_First_Participant_ID, sample_id)
-```
-
-# Print Out Tables
+             !(BRAF.V600E & fourty_or_older) & (CTNNB1.Exon3 & !fourty_or_older) ~ "CRANIO, ADAM", 
```

Comment:
Contradictory to the description in README. In README, 
```
Adamantinomatous craniopharyngiomas occur mostly in childhood or young adolescence (0-39 years), but can be seen in adults
```
In script, it excluded all potential patients carrying CTNNB1 mutation and older than 40.

Reply:
I agree with @zzgeng's comment.

Reply:
yes, thank you - I removed this logic too quickly when revamping!